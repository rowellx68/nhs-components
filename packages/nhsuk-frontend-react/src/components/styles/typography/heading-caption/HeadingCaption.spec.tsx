import React from 'react';
import { it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { composeStory } from '@storybook/react';
import meta, { XL as HeadingCaptionStory } from './HeadingCaption.stories';

const HeadingCaption = composeStory(HeadingCaptionStory, meta);

it.each(['xl', 'l', 'm'] as const)(
  `should render the %s HeadingCaption component`,
  (size) => {
    const { container } = render(<HeadingCaption size={size} />);

    expect(
      container.querySelector(`.nhsuk-caption-${size}`),
    ).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  },
);
