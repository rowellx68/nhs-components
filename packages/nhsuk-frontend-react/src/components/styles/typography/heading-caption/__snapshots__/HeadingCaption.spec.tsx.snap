// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render the l HeadingCaption component 1`] = `
<div>
  <span
    class="nhsuk-caption-l"
  >
    nhsuk-caption-
    l
  </span>
</div>
`;

exports[`should render the m HeadingCaption component 1`] = `
<div>
  <span
    class="nhsuk-caption-m"
  >
    nhsuk-caption-
    m
  </span>
</div>
`;

exports[`should render the xl HeadingCaption component 1`] = `
<div>
  <span
    class="nhsuk-caption-xl"
  >
    nhsuk-caption-
    xl
  </span>
</div>
`;
